identity/src/commonMain/kotlin/com/android/identity/nfc/NdefRecord.kt

```diff
@@ -68,14 +72,14 @@ data class NdefRecord(
     }
 
     /**
-     * Encodes the record into a [Buffer].
+     * Encodes the record into a [ByteStringBuilder].
      *
-     * @param buf the [Buffer] to append the encoding of the record to.
+     * @param bsb the [ByteStringBuilder] to append the encoding of the record to other records.
      * @param isMessageBegin set to `true` if this is the first record of a message.
      * @param isMessageEnd set to `true` if this is the last record of a message.
      */
     internal fun encode(
-        buf: Buffer,
+        bsb: ByteStringBuilder,
```

Comment:
This leads to some new mixing between ByteArray and ByteString (and ByteStringBuilder), and this intermediate state has its own issues converting back and forth. If we need to make this Buffer -> ByteStringBuilder conversion, I'm wondering if it'd be cleaner if the downstream data types also switched from ByteArray to ByteString? ...Or if that's going to be a huge amount of additional work?